src/robot/libraries/Collections.py

```diff
@@ -288,10 +303,9 @@ def reverse_list(self, list_):
         =>
         | ${L3} = ['c', 'b', 'a']
         """
-        self._validate_list(list_)
         list_.reverse()
 
-    def sort_list(self, list_):
+    def sort_list(self, list_: list):
```

Comment:
It's annoying that we cannot use `MutableSequence` because it doesn't have `sort()`. If we wanted to support custom sequences that have it, typing could be `MutableSequence` and then code could do something like:
```python
if not hasattr(list_, 'sort'):
    list_ = list(list_)
```
but that's probably not worth the effort at least now.